File: VB/WinFormsApp/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Controls

Namespace WinFormsApp
	Partial Public Class Form1
		Inherits Form
        Public Function CreateDataSource() As List(Of MyTestClass)
            Dim myList As New List(Of MyTestClass)()
            myList.Add(New MyTestClass() With {.ID = 0, .Value = "AA1A"})
            myList.Add(New MyTestClass() With {.ID = 1, .Value = "AA"})
            myList.Add(New MyTestClass() With {.ID = 2, .Value = "AAA"})
            myList.Add(New MyTestClass() With {.ID = 3, .Value = "D"})
            myList.Add(New MyTestClass() With {.ID = 4, .Value = "E"})

            Return myList
        End Function

		Public Sub New()
			InitializeComponent()

		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim rep As RepositoryItemCustomridLookUpEdit = customGridLookUpEdit1.Properties

			rep.DataSource = CreateDataSource()
			rep.DisplayMember = "Value"
			rep.ValueMember = "Value"
		End Sub

	End Class

    Public Class MyTestClass
        Private privateID As Integer
        Public Property ID() As Integer
            Get
                Return privateID
            End Get
            Set(ByVal value As Integer)
                privateID = value
            End Set
        End Property
        Private privateValue As String
        Public Property Value() As String
            Get
                Return privateValue
            End Get
            Set(ByVal value As String)
                privateValue = value
            End Set
        End Property

    End Class
End Namespace

File: CS/WinFormsApp/Form1.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.XtraEditors.Controls;

namespace WinFormsApp
{
    public partial class Form1 : Form
    {
        public List<MyClass> CreateDataSource()
        {
            List<MyClass> list = new List<MyClass>();
            list.Add(new MyClass() { ID = 0, Value = "AA1A" });
            list.Add(new MyClass() { ID = 1, Value = "AA" });
            list.Add(new MyClass() { ID = 2, Value = "AAA" });
            list.Add(new MyClass() { ID = 3, Value = "D" });
            list.Add(new MyClass() { ID = 4, Value = "E" });

            return list;
        }

        public Form1()
        {
            InitializeComponent();

        }

        private void Form1_Load(object sender, EventArgs e)
        {
            RepositoryItemCustomridLookUpEdit rep = customGridLookUpEdit1.Properties;

            rep.DataSource = CreateDataSource();
            rep.DisplayMember = "Value";
            rep.ValueMember = "Value";            
        }

    }

    public class MyClass
    {
        public int ID { get; set; }
        public string Value { get; set; }
       
    }
}
